crates/hyperswitch_connectors/src/connectors/jpmorgan/transformers.rs

```diff
@@ -1,52 +1,133 @@
+    pub grant_type: String,
+    pub scope: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct JpmorganAuthUpdateResponse {
+    pub access_token: Secret<String>,
+    pub scope: String,
+    pub token_type: String,
+    pub expires_in: i64,
+}
+
+impl TryFrom<&RefreshTokenRouterData> for JpmorganAuthUpdateRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_item: &RefreshTokenRouterData) -> Result<Self, Self::Error> {
+        Ok(Self {
+            grant_type: String::from("client_credentials"),
+            scope: String::from("jpm:payments:sandbox"),
+        })
+    }
+}
+
+impl<F, T> TryFrom<ResponseRouterData<F, JpmorganAuthUpdateResponse, T, AccessToken>>
+    for RouterData<F, T, AccessToken>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: ResponseRouterData<F, JpmorganAuthUpdateResponse, T, AccessToken>,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            response: Ok(AccessToken {
+                token: item.response.access_token,
+                expires: item.response.expires_in,
+            }),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
 pub struct JpmorganPaymentsRequest {
-    amount: StringMinorUnit,
-    card: JpmorganCard,
+    capture_method: CapMethod,
+    amount: MinorUnit,
+    currency: common_enums::Currency,
+    merchant: JpmorganMerchant,
+    payment_method_type: JpmorganPaymentMethodType,
 }
 
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
 pub struct JpmorganCard {
-    number: cards::CardNumber,
-    expiry_month: Secret<String>,
-    expiry_year: Secret<String>,
-    cvc: Secret<String>,
-    complete: bool,
+    account_number: Secret<String>,
+    expiry: Expiry,
+    is_bill_payment: bool,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JpmorganPaymentMethodType {
+    card: JpmorganCard,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Expiry {
+    month: Secret<String>,
+    year: Secret<String>,
+}
+
+#[derive(Serialize, Debug, Default, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JpmorganMerchantSoftware {
+    company_name: String,
+    product_name: String,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JpmorganMerchant {
+    merchant_software: JpmorganMerchantSoftware,
+}
+
+fn map_capture_method(capture_method: CaptureMethod) -> CapMethod {
+    match capture_method {
+        CaptureMethod::Automatic => CapMethod::Now,
+        CaptureMethod::Manual | CaptureMethod::ManualMultiple => CapMethod::Manual,
+        CaptureMethod::Scheduled | CaptureMethod::SequentialAutomatic => CapMethod::Delayed,
+    }
```

Comment:
Only map the capture methods that you actually support

Reply:
Done